[vcpkg export] manifest mode (#116)
microsoft/vcpkg-docs: vcpkg/commands/export.md

## Description

The `export` command exports built packages from the [installed directory](common-options.md#install-root) into a standalone developer SDK.

`export` produces a standalone, distributable SDK (Software Development Kit) that can be used on another machine without separately acquiring vcpkg. It contains:

1. The prebuilt binaries for the selected packages
2. Their transitive dependencies
3. [Integration files](#standard-integration), such as a [CMake toolchain][cmake] or [MSBuild props/targets][msbuild]

[!NOTE]

> This command behavior is different in [Classic Mode](../users/classic-mode.md) and [Manifest Mode](../users/manifests.md)

Reviewer: "command's behavior"

The `export` command does not install any packages or transitive dependencies. It only exports packages that are already installed. 

Refer to [Manifest Mode](export.md#Manifest-Mode) or [Classic Mode](export.md#Classic-Mode) for more details.

## <a name="Classic-Mode"></a>Classic Mode

In classic mode, `vcpkg export` accepts [triplet-qualified package specification](install.md#package-syntax) arguments (for example: `zlib:x64-windows`)

> When using `vcpkg export` in classic mode, you explicitly specify the packages to export by adding `<portname:triplet>` arguments to the command line. Each argument corresponds to a specific package and its configuration (defined by the port and triplet).

Reviewer 2: ```suggestion
You specify packages to export by adding `<port name>:<triplet>` arguments to the command line.
```

The second statement seems redundant to me.

For example, to export the `sqlite` package for `x64-windows` and `x64-linux`, use:

```no-highlight
vcpkg export sqlite:x64-windows sqlite:x64-linux --zip
```

This command exports the specified packages in zip format. Both `sqlite:x64-windows` and `sqlite:x64-linux` must be installed prior to running `vcpkg export`.

## <a name="Manifest-Mode"></a>Manifest Mode

> In manifest mode, the `vcpkg export` command automatically exports everything currently installed according to the specifications in the manifest file (`vcpkg.json`). This streamlined approach embodies the philosophy of manifest mode, where the entire state of installed packages is determined by the manifest file.

Reviewer 2: ```suggestion
In manifest mode, the `vcpkg export` command exports everything currently installed according to the dependencies declared in the manifest file (`vcpkg.json`). This approach embodies the philosophy of manifest mode, where the entire state of installed packages is determined by the manifest file.
```
 I think this needs a rewrite, since the packages in `vcpkg_installed` can actually differ from the packages declared in `vcpkg.json`.
 
> In manifest mode, the command exports all currently installed packages. The installation directory includes all the packages declared in the manifest (`vcpkg.json`) as well as their transitive dependencies. Run `vcpkg install` before using this command to ensure that all required packages are installed.

> This mode has a few important implications for how you use `vcpkg export`. Firstly, unlike in classic mode, you cannot specify individual `<portname:triplet>` arguments for the `vcpkg export` command. The command operates on the entire installed state as a single unity, respecting the dependencies and versions specified in the manifest file.

Reviewer: "single entity"

> This mode has a few important implications for how you use `vcpkg export`. Firstly, unlike in classic mode, you cannot specify individual `<portname:triplet>` arguments for the `vcpkg export` command. The command operates on the entire installed state as a single entity, respecting the dependencies and versions specified in the manifest file.

Reviewer 2: ```suggestion
In this mode you can't specify individual packages to export. The command operates on the entire installed state as a single entity, respecting the dependencies and versions specified in the manifest file.
```

> Another key difference in manifest mode is the requirement of the `--output-dir` option. This option specifies the directory where the exported packages will be stored, and you must provide it when using `vcpkg export` in manifest mode. This is different from classic mode, where `vcpkg` selects a default output directory (`vcpkg-root`) if one is not provided.

Reviewer 2: ```suggestion
Another difference is the requirement of the `--output-dir` option. This option specifies the directory where the exported packages will be stored.
```

For example, from the manifest directory
```no-highlight
vcpkg export --zip --output-dir=.\exports
```

> exports all currently installed packages to the `.\exports` directory and compresses them into a zip file. The `--zip` option specifies that the exported packages should be compressed into a zip file.

Reviewer 2: ```suggestion
Exports all currently installed packages to the `.\exports` directory in a zip file.
```

Specifies the output directory.

> All top-level SDK files will be produced into this directory. This option is mandatory in manifest mode. In classic mode, this option is not required and defaults to the [vcpkg root directory](../users/config-environment.md#vcpkg_root).

Reviewer 2: ```suggestion
All top-level SDK files will be produced into this directory. This option is required in manifest mode. In classic mode, this is optional and defaults to the [vcpkg root directory](../users/config-environment.md#vcpkg_root).
```
